src/import/FSHImporter.ts

```diff
@@ -550,6 +631,217 @@ export class FSHImporter extends FSHVisitor {
+          [codePart, description] = code
+            .match(/"([^\s\\"]|\\"|\\\\)+(\s([^\s\\"]|\\"|\\\\)+)*"/g)
+            .map(quotedString => quotedString.slice(1, -1));
+        } else {
+          // codePart is not a quoted string.
+          // if there is a description after the code,
+          // it will be separated by whitespace before the leading "
+          const codeEnd = code.match(/\s+"/)?.index;
+          if (codeEnd) {
+            codePart = code.slice(0, codeEnd);
+            description = code
+              .slice(codeEnd)
+              .trim()
+              .slice(1, -1);
+          } else {
+            codePart = code.trim();
+          }
+        }
+        concepts.push(
+          new FshCode(codePart, from.system, description).withLocation(location).withFile(this.file)
+        );
+      });
+    }
+    return [concepts, from];
+  }
+
+  visitVsFilterComponent(
+    ctx: pc.VsFilterComponentContext
+  ): [ValueSetFilter[], ValueSetComponentFrom] {
+    const filters: ValueSetFilter[] = [];
+    const from: ValueSetComponentFrom = ctx.vsComponentFrom()
+      ? this.visitVsComponentFrom(ctx.vsComponentFrom())
+      : {};
+    if (ctx.vsFilterList()) {
+      ctx
+        .vsFilterList()
+        .vsFilterDefinition()
+        .forEach(filterDefinition => {
+          try {
+            filters.push(this.visitVsFilterDefinition(filterDefinition));
+          } catch (e) {
+            logger.error(e, {
+              location: this.extractStartStop(filterDefinition),
+              file: this.file
+            });
+          }
+        });
+    }
+    return [filters, from];
+  }
+
+  visitVsComponentFrom(ctx: pc.VsComponentFromContext): ValueSetComponentFrom {
+    const from: ValueSetComponentFrom = {};
+    if (ctx.vsFromSystem()) {
+      from.system = ctx
+        .vsFromSystem()
+        .SEQUENCE()
+        .getText();
+    }
+    if (ctx.vsFromValueset()) {
+      if (ctx.vsFromValueset().SEQUENCE()) {
+        from.valueSets = [
+          ctx
+            .vsFromValueset()
+            .SEQUENCE()
+            .getText()
+        ];
+      } else if (ctx.vsFromValueset().COMMA_DELIMITED_SEQUENCES()) {
+        from.valueSets = ctx
+          .vsFromValueset()
+          .COMMA_DELIMITED_SEQUENCES()
+          .getText()
+          .split(',')
+          .map(fromVs => fromVs.trim());
+      }
+    }
+    return from;
+  }
+
+  /**
+   * The replace makes FSH permissive in regards to the official specifications,
+   * which spells operator "descendant-of" as "descendent-of".
+   * @see {@link http://hl7.org/fhir/valueset-filter-operator.html}
+   */
+  visitVsFilterDefinition(ctx: pc.VsFilterDefinitionContext): ValueSetFilter {
+    const property = ctx
+      .SEQUENCE()
+      .getText()
+      .toLocaleLowerCase() as VsProperty;
+    if (Object.values(VsProperty).indexOf(property) < 0) {
+      throw new ValueSetFilterPropertyError(ctx.SEQUENCE().getText());
```

Comment:
I know this has already been merged, but I agree with the second thoughts that @mint-thompson expressed earlier -- we should be permissive here and allow any property (not just the four).  If FHIR intended for only four possible values, it would have bound `property` with a `required` value set containing only those four values.  Since it did not bind `property` at all, we should assume that means that anything that is a valid `code` type is allowed.